homeassistant/components/traccar/device_tracker.py

```diff
@@ -105,10 +153,35 @@ def __init__(self, api, hass, async_see, scan_interval, custom_attributes):
                 attr[ATTR_BATTERY_LEVEL] = device['battery']
             if device.get('motion') is not None:
                 attr[ATTR_MOTION] = device['motion']
+            if device.get('traccar_id') is not None:
+                attr[ATTR_TRACCAR_ID] = device['traccar_id']
             for custom_attr in self._custom_attributes:
                 if device.get(custom_attr) is not None:
                     attr[custom_attr] = device[custom_attr]
             await self._async_see(
                 dev_id=slugify(device['device_id']),
                 gps=(device.get('latitude'), device.get('longitude')),
                 attributes=attr)
+
+    async def import_events(self):
+        """Import events from Traccar."""
+        device_ids = [device['id'] for device in self._api.devices]
+        end_interval = datetime.utcnow()
+        start_interval = end_interval - self._scan_interval
+        events = await self._api.get_events(device_ids=device_ids,
+                                            from_time=start_interval,
+                                            to_time=end_interval,
+                                            event_types=self._event_types)
+        if events is not None:
+            for event in events:
+                device_name = list(filter(lambda dev:
+                                          dev.get('id') == event['deviceId'],
+                                          self._api._devices)
+                                   )[0].get('name')
```

Comment:
We can use a generator expression and `next` instead since we're only interested in one item.

```py
name = ((
    dev.get('name) for dev in self._api._devices
    if dev.get('id') == event['deviceId']), None)